pkg/pool-weighted/test/LiquidityBootstrappingPool.test.ts

```diff
@@ -1,2 +1,161 @@
+import { ethers } from 'hardhat';
+import { expect } from 'chai';
+import { BigNumber } from 'ethers';
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/dist/src/signer-with-address';
+import { actionId } from '@balancer-labs/v2-helpers/src/models/misc/actions';
+import { fp } from '@balancer-labs/v2-helpers/src/numbers';
+import { MINUTE, advanceTime, currentTimestamp } from '@balancer-labs/v2-helpers/src/time';
+
+import TokenList from '@balancer-labs/v2-helpers/src/models/tokens/TokenList';
+import WeightedPool from '@balancer-labs/v2-helpers/src/models/pools/weighted/WeightedPool';
+
+import { itBehavesAsWeightedPool } from './BaseWeightedPool.behavior';
+
 // eslint-disable-next-line @typescript-eslint/no-empty-function
-describe('LiquidityBootstrappingPool', function () {});
+describe('LiquidityBootstrappingPool', function () {
+  describe('as a 2-token weighted pool', () => {
+    itBehavesAsWeightedPool(2);
+  });
+
+  describe('as a 3-token weighted pool', () => {
+    itBehavesAsWeightedPool(3);
+  });
+
+  let trader: SignerWithAddress, lp: SignerWithAddress, owner: SignerWithAddress;
+
+  before('setup signers', async () => {
+    [, lp, trader, owner] = await ethers.getSigners();
+  });
+
+  let tokens: TokenList;
+
+  sharedBeforeEach('deploy tokens', async () => {
+    tokens = await TokenList.create(['MKR', 'DAI', 'SNX', 'BAT'], { sorted: true });
+    await tokens.mint({ to: [lp, trader], amount: fp(200) });
+  });
+
+  let pool: WeightedPool;
+  const weights = [fp(0.3), fp(0.55), fp(0.1), fp(0.05)];
+  const initialBalances = [fp(0.9), fp(1.8), fp(2.7), fp(3.6)];
+
+  sharedBeforeEach('deploy pool', async () => {
+    const params = { tokens, weights, owner, lbp: true, swapEnabledOnStart: true };
```

Comment:
For now this might be fine, but it's weird that `swapEnabledOnStart` does nothing if `lbp` is not set. Maybe we could have a `type` field that can be set to `LBP`, and then associated settings? e.g.

```
type LBPConfig = {
    kind: 'LBP';
    swapEnabled?: boolean;
}

type OracleConfig = {
    kind: 'Oracle';
    duration?: Number;
}

type MiscConfig = LBPConfig | OracleConfig;
```

(I'm using oracle as an example of another 'variant' - I'd start with just LBP for now)

Reply:
Oracle is another type - oracleEnabled also does nothing if twoTokens isn't set. I was trying not to make a new model.